src/elements/checkbox.js

```diff
@@ -28,10 +62,52 @@ class Checkbox extends Component {
         }
     }
 
+    onClick(event) {
+        const { disabled, id, onChange } = this.props;
+        const { isChecked } = this.state;
+
+        if (!disabled) {
+            if (!_.isUndefined(onChange)) {
+                onChange(id, !isChecked, event);
+            } else {
+                this.setState({
+                    isChecked: !isChecked,
+                }, () => {
+                    this._inputRef.checked = !isChecked;
+                });
+            }
+        } else {
+            event.stopPropagation();
+        }
+    }
+
+    onLabelClick(event) {
+        const { labelClick } = this.props;
+
+        if (!_.isUndefined(labelClick) && labelClick === false) {
```

Comment:
I think `labelClick === false` is sufficient here; checking it for `undefined` beforehand doesn't really add anything, does it?  NAB of course.

Reply:
Yep, I agree.